src/main/kotlin/com/asarkar/spring/test/redis/SocketUtils.kt

```diff
@@ -0,0 +1,153 @@
+/*
+ * Copyright 2002-2020 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.asarkar.spring.test.redis
+
+import org.springframework.util.Assert
+import java.net.DatagramSocket
+import java.net.InetAddress
+import java.net.ServerSocket
+import java.util.Random
+import javax.net.ServerSocketFactory
+
+/**
+ * Simple utility methods for working with network sockets  for example,
+ * for finding available ports on `localhost`.
+ *
+ * Within this class, a TCP port refers to a port for a [ServerSocket];
+ * whereas, a UDP port refers to a port for a [DatagramSocket].
+ *
+ * org.springframework.util.SocketUtils was removed in Spring 6.
+ * From Springboot 3.2.3 version, SocketUtils cannot be used.
+ * org.springframework.test.util.TestSocketUtils can be used as an alternative, but embedded-redis-spring has to support older versions, so org.springframework.util.SocketUtils was copied.
+ *
+ * @author Sam Brannen
+ * @author Ben Hale
+ * @author Arjen Poutsma
+ * @author Gunnar Hillert
+ * @author Gary Russell
+ * @author fennec-fox
+ * @since 4.0
+ */
```

Comment:
Please remove spurious author names and put your name. Also, update since version.